src/Infrastructure.VS/Roslyn/SolutionRoslynAnalyzerManager.cs

```diff
@@ -160,42 +139,45 @@ private void UpdateAnalyzersIfChanged(ImmutableArray<AnalyzerFileReference> anal
             return;
         }
 
-        UpdateAnalyzers(analyzersToUse);
+      
+        RemoveCurrentAnalyzers();
+        AddAnalyzer(analyzersToUse); 
     }
 
     private bool DidAnalyzerChoiceChange(ImmutableArray<AnalyzerFileReference> analyzersToUse)
     {
-        Debug.Assert(currentAnalyzers is not null);
         return !analyzerComparer.Equals(currentAnalyzers, analyzersToUse);
     }
 
-    private void UpdateAnalyzers(ImmutableArray<AnalyzerFileReference> analyzersToUse)
+    private void RemoveCurrentAnalyzers()
     {
-        if (!roslynWorkspace.TryApplyChanges(roslynWorkspace.CurrentSolution.RemoveAnalyzerReferences(currentAnalyzers!.Value)))
+        if (currentAnalyzers.HasValue && !roslynWorkspace.TryApplyChanges(roslynWorkspace.CurrentSolution.RemoveAnalyzerReferences(currentAnalyzers.Value)))
         {
-            const string message = "Failed to remove analyzer references while updating analyzers";
-            Debug.Assert(true, message);
-            logger.LogVerbose(message);
+            logger.LogVerbose(Resources.RoslynAnalyzersNotRemoved);
             throw new NotImplementedException();
```

Comment:
Can you fix this as well?

Reply:
You mean to throw a different exception? Or maybe no exception at all.